Add reset HP option to creature menu

Trantion897 opened the pull request:
## Pull Request Description

Adds an option to reset HP for creatures where the HP is not equal to current_max
Uses the "undo" icon for this, and also changes the icon for the existing "Reset AC" option to "undo" because I think using the HP icon twice in the same menu was potentially confusing

## Changes Proposed

- [ ] Add option to reset creature HP
- [ ] Change Reset AC option to use "undo" icon

## Related Issues

None

## Checklist

<!-- Make sure to check the items below before submitting your pull request -->

- [X] I have read the contribution guidelines and code of conduct.
- [X] I have tested the changes locally and they are working as expected. (see Additional Notes)
- [X] I have added appropriate comments and documentation for the code changes.
- [X] My code follows the coding style and standards of this project.
- [X] I have rebased my branch on the latest main (or master) branch.
- [X] All tests (if applicable) have passed successfully.
- [X] I have run linters and fixed any issues.
- [X] I have checked for any potential security issues or vulnerabilities.

## Screenshots (if applicable)

![image](https://github.com/user-attachments/assets/eca0f78c-6dc5-4b09-b1da-a91e6105bdf2)

## Additional Notes

Tested:
* Using PC and NPC
* Reset HP option does not appear at full health, or full health + temporary HP
* Reset HP option appears when below full health
* Reset HP option appears when below full health + temporary HP, and correctly resets to full health + temporary HP
* Reset HP option respects changes to max HP, including resetting to changed max HP

SondreElg commented:
I do like the change, but I'm unsure if we want to reset HP to `creature.current_max`, as that value can be affected by changes to max HP. It might be nice in the event you have players under the effect of something like Heroes' Feast, and want to reset their damage taken without resetting the change to their max HP. However, I do imagine a most common use of this effect being someone healing damage done to a player's max HP

We might actually want to provide two options, in the form of a "hard" and a "soft" HP reset option. If we do provide that though

valentine195 commented:
> We might actually want to provide two options, in the form of a "hard" and a "soft" HP reset option to account for both scenarios. If we do provide that though, we should probably add that as an alternative in the encounter menu as well.

I agree, and I would recommend making HP open a submenu.

Trantion897 commented:
You're right, you would usually want to reset max HP at the same time as HP (at least in D&D, and probably PF2 - not sure about all other systems). Looking a bit more, the existing Reset HP/Status option in the encounter menu does what I was looking for, which is resetting the PCs after they've healed & rested during a session. 

I have implemented both options on the creature menu and pushed, although rebasing onto main seems to have messed up the PR - let me know if I should create a new one. Would you like the same option(s) on the encounter menu too, or is that basically the same as Reset HP/Status? I think there are a few statuses that don't go away on a long rest, like D&D's Exhaustion and PF2e's Doomed, but I'd be tracking those outside of an initiative tracker anyway

![image](https://github.com/user-attachments/assets/56913cf3-afc9-4ee8-82c5-856f7aa9be26)

SondreElg commented:
Looks great!

It would be nice to have both options in the encounter menu as well though. Effects that reduce max HP until a long rest aren't that uncommon in 5e, not to mention other systems, but a short rest for players usually means they all heal to their current max. 

It'd be nice to be able to heal everyone at once instead of doing it one player at a time.

SondreElg commented:
Hey, have you had time to look at this, and do you have any questions you'd like my help with?

Trantion897 commented:
I've pushed a version with various different options to reset current HP, max HP and statuses. AC is always reset as before - I didn't want to make the submenu too big. But if you have any other suggestions, let me know. I also changed the reset icon in the per-creature menu to be the same as in the encounter menu.

![image](https://github.com/user-attachments/assets/dcd77f0f-81c4-4c48-a5ef-eb1b828f13ba)

I also threw in a couple of small UI changes. Firstly, creatures where the max HP has been changed are highlighted in the same way that creatures with temporary HP already were:

![image](https://github.com/user-attachments/assets/2d266ccb-1023-4ce1-b89f-9d03285d99c8)

Secondly, while testing this I kept thinking that the 'm' prefix in the apply damage *sets* the current_max HP instead of reducing it. So I changed the label to make it clearer:

![image](https://github.com/user-attachments/assets/1b896b64-fb51-4d21-b110-15175d187a82)

These are both separate commits if you don't want either of them.
